### PR TITLE
MRC-37 Distributions

### PR DESCRIPTION
**Pull request by adamjw3**

### CODE REVIEW

File: .bsp/sbt.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"name":"sbt","version":"1.6.1","bspVersion":"2.0.0-M5","languages":["scala"],"argv":["/Library/Java/JavaVirtualMachines/jdk1.8.0_211.jdk/Contents/Home/jre/bin/java","-Xms100m","-Xmx100m","-classpath","/Users/adamwright/Library/Application Support/JetBrains/IdeaIC2022.1/plugins/Scala/launcher/sbt-launch.jar","xsbt.boot.Boot","-bsp","--sbt-launch-jar=/Users/adamwright/Library/Application%20Support/JetBrains/IdeaIC2022.1/plugins/Scala/launcher/sbt-launch.jar"]}
```
Comment (girishkamat) [inline .bsp/sbt.json:1]: Please remove this file

File: app/models/Distribution.scala [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  case object Yes extends WithName("yes") with Distribution
+  case object No extends WithName("no") with Distribution
+
+  val values: Seq[Distribution] = Seq(
+    Yes,
+    No
+  )
+
+  def options(implicit messages: Messages): Seq[RadioItem] = values.zipWithIndex.map { case (value, index) =>
```
Comment (girishkamat) [inline app/models/Distribution.scala:35]: Is this options method used?

File: app/navigation/Navigator.scala [hunk 3]
```diff
@@ -47,6 +49,13 @@ class Navigator @Inject() () {
     _ => routes.CheckYourAnswersController.onPageLoad
   }
 
+  def complexRoute(answers: UserAnswers): Call =
```
Comment (girishkamat) [inline app/navigation/Navigator.scala:52]: This is very generic name. Can we rename it - distributionsNextRoute

File: app/views/DistributionView.scala.html [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    @formHelper(action = routes.DistributionController.onSubmit(mode), 'autoComplete -> "off") {
+
+        @if(form.errors.nonEmpty) {
+            @govukErrorSummary(ErrorSummaryViewModel(form,  errorLinkOverrides = Map("value" -> "value_0")))
+        }
+
+        <h1 class="govuk-fieldset__legend--l">
+            @messages("distribution.heading")
+        </h1>
+        <p class="govuk-body govuk-!-margin-bottom-4"></p>
```
Comment (girishkamat) [inline app/views/DistributionView.scala.html:38]: Do we need the element above?

File: app/views/DistributionsIncludedView.scala.html [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            summary = Text(messages("distributionsIncluded.detailsTitle")),
+            content = HtmlContent(messages("distributionsIncluded.detailsHtml"))
+        ))
+
+        @govukRadios(
+            RadiosViewModel(
+                field  = form("distributionsIncluded"),
+                legend = LegendViewModel(messages("distributionsIncluded.heading")).withCssClass("govuk-visually-hidden"),
+                items = Seq(
+                    RadioItem(
```
Comment (girishkamat) [inline app/views/DistributionsIncludedView.scala.html:70]: Remove the label value please. For the label, can we use a generic yes/no e.g site.yes = Yes

File: conf/messages.en [hunk 6]
```diff
@@ -112,7 +112,7 @@ profit.change.hidden = Taxable profit
 distribution.title = Distributions
 distribution.heading = Distributions
 distribution.checkYourAnswersLabel = Distributions
-distribution.error.nonNumeric = Enter your Distributions using numbers
+sdistribution.error.nonNumeric = Enter your Distributions using numbers
```
Comment (girishkamat) [inline conf/messages.en:115]: typo?

File: conf/messages.en [hunk 7]
```diff
@@ -147,3 +147,41 @@ taxableProfit.chargeableProfitsOnCTR = Chargeable profits for Corporation Tax on
 taxableProfit.whereCanIFindMyCompanyTaxProfit = Where can I find my company''s taxable profit?
 taxableProfit.youCanFindYourCompanyProfitVia = You can find your company's profit via your accountant or accounting software
 taxableProfit.hint = Must be entered as a whole number. For example, 70000
+
+distribution.title = Did your company receive any distributions?
```
Comment (girishkamat) [inline conf/messages.en:151]: move this to the distribution section above

File: test/navigation/NavigatorSpec.scala [hunk 8]
```diff
@@ -34,14 +34,14 @@ class NavigatorSpec extends SpecBase {
           .onPageLoad(NormalMode)
       }
 
-      "must go from TaxableProfit page to InputScreen page" in {
-        navigator.nextPage(TaxableProfitPage, NormalMode, UserAnswers("id")) mustBe routes.InputScreenController
+      "must go from TaxableProfit page to Distribution page" in {
+        navigator.nextPage(TaxableProfitPage, NormalMode, UserAnswers("id")) mustBe routes.DistributionController
           .onPageLoad(NormalMode)
       }
```
Comment (girishkamat) [inline test/navigation/NavigatorSpec.scala:41]: Where is the test to go from Distribution page to Distributions Included page?

File: test/pages/DistributionsIncludedPageSpec.scala [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+package pages
+
+import models.DistributionsIncluded
+import pages.behaviours.PageBehaviours
+
+class DistributionsIncludedSpec extends PageBehaviours {
+
+  "DistributionsIncludedPage" - {
+
+    /*beRetrievable[DistributionsIncluded](DistributionsIncludedPage)
```
Comment (girishkamat) [inline test/pages/DistributionsIncludedPageSpec.scala:26]: why are these tests commented?

File: .bsp/sbt.json (hunk 1 above)
Comment (adamjw3) [reply .bsp/sbt.json:1]: done

File: app/models/Distribution.scala (hunk 2 above)
Comment (adamjw3) [reply app/models/Distribution.scala:35]: Thats being used

File: app/navigation/Navigator.scala (hunk 3 above)
Comment (adamjw3) [reply app/navigation/Navigator.scala:52]: done

File: app/views/DistributionView.scala.html (hunk 4 above)
Comment (adamjw3) [reply app/views/DistributionView.scala.html:38]: Removed, was in the Prototype

File: app/views/DistributionsIncludedView.scala.html (hunk 5 above)
Comment (adamjw3) [reply app/views/DistributionsIncludedView.scala.html:70]: Done

File: test/pages/DistributionsIncludedPageSpec.scala (hunk 9 above)
Comment (adamjw3) [reply test/pages/DistributionsIncludedPageSpec.scala:26]: Added back in

File: conf/messages.en (hunk 6 above)
Comment (adamjw3) [reply conf/messages.en:115]: This has been removed and merged with the below

File: conf/messages.en (hunk 7 above)
Comment (adamjw3) [reply conf/messages.en:151]: see above

File: test/navigation/NavigatorSpec.scala (hunk 8 above)
Comment (adamjw3) [reply test/navigation/NavigatorSpec.scala:42]: Test added

File: app/views/DistributionsIncludedView.scala.html [hunk 10]
```diff
@@ -69,12 +69,12 @@ <h1 class="govuk-fieldset__legend--l">
...
                     RadioItem(
                         content = Text(messages("distributionsIncluded.option1Label")),
-                        value = Some(messages("distributionsIncluded.option1Value")),
+                        value = Some(messages("site.yes")),
                         conditionalHtml = Some(yesHtml)
                     ),
                     RadioItem(
                         content = Text(messages("distributionsIncluded.option2Label")),
-                        value = Some(messages("distributionsIncluded.option2Value"))
+                        value = Some(messages("site.No"))
```
Comment (girishkamat) [inline app/views/DistributionsIncludedView.scala.html:77]: site.No will that work if its upper case?

Comment (girishkamat) [reply app/views/DistributionsIncludedView.scala.html:77]: Also, the values is constant across languages. You should be using site.yes and site.not for content, not the value. Value should be hardcoded, regardless of the language
 value = Some("yes")

value = Some("no")

File: test/navigation/NavigatorSpec.scala (hunk 8 above)
Comment (girishkamat) [reply test/navigation/NavigatorSpec.scala:42]: Please add as table for all scenarios

 UserAnswers("id").set(DistributionPage, Distribution.Yes)
 UserAnswers("id").set(DistributionPage, Distribution.No)
UserAnswers("id").set(DistributionPage, "unknown")

Comment (adamjw3) [reply test/navigation/NavigatorSpec.scala:42]: Done

Comment (adamjw3) [reply test/navigation/NavigatorSpec.scala:42]: Done

**Review by girishkamat — APPROVED** [review]

**Pull request merged**